arch/inst/C/c.sd.yaml

```diff
@@ -32,9 +33,17 @@ access:
   vu: always
 operation(): |
   if (implemented?(ExtensionName::C) && (CSR[misa].C == 1'b0)) {
-    raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+      raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
   }
 
   XReg virtual_address = X[rs1] + imm;
 
-  write_memory<64>(virtual_address, X[rs2], $encoding);
+  if (XLEN == 32) {
+      if (implemented?(ExtensionName::Zclsd)) {
+          write_memory<64>(virtual_address, X[rs2], $encoding);
```

Comment:
Now that the XLEN64 and XLEN32/Zclsd implementations are merged, we can't use the `not` attribute to restrict the allowed registers to be even for Zclsd, so we'll need to verify that `rs2` is even, and we need to gather the data from both 32-bit registers in the register pair as you had previously:
```
  Bits<64> data = {X[rs2+1], X[rs2]};
  write_memory<64>(eff_addr, data, $encoding);
```